sdk/search/Azure.Search.Documents/src/Models/QueryCaption.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    /// <summary>
+    /// Configuration for how semantic search captions search results.
+    /// </summary>
+    public partial class QueryCaption
+    {
+        private const string QueryCaptionRawSplitter = "|highlight-";
+
+        /// <summary>
+        /// A value that specifies whether <see cref="SemanticSearchResult.Captions"/> should be returned as part of the search response.
+        /// <para>The default value is <see cref="QueryCaptionType.None"/>.</para>
+        /// </summary>
+        public QueryCaptionType? CaptionType { get; set; }
+
+        /// <summary>
+        /// If <see cref="CaptionType"/> is set to <see cref="QueryCaptionType.Extractive"/>, setting this to <c>true</c> enables highlighting of the returned captions.
+        /// It populates <see cref="CaptionResult.Highlights"/>.
+        /// <para>The default value is <c>true</c>.</para>
+        /// </summary>
+        public bool? HighlightEnabled { get; set; }
```

Comment:
Does a nullable bool make sense or would `public bool HighlightEnabled { get; set; } = true` make more sense?

Reply:
It will be null in case of non semantic search.

Reply:
@tg-msft I'm creating an instance [here](https://github.com/Azure/azure-sdk-for-net/blob/6593a9223374af7e946c89d776cc62ad624060e2/sdk/search/Azure.Search.Documents/src/Options/SemanticSearchOptions.cs#L19) so that I can utilize the generated serialization and deserialization code, which accesses [these](https://github.com/Azure/azure-sdk-for-net/blob/6593a9223374af7e946c89d776cc62ad624060e2/sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs#L185-L189) properties.

Reply:
I've set default to true.